Source/WebKit/GPUProcess/graphics/RemoteGraphicsContextGL.cpp

```diff
@@ -106,8 +106,8 @@ RemoteGraphicsContextGL::~RemoteGraphicsContextGL()
 void RemoteGraphicsContextGL::initialize(GraphicsContextGLAttributes&& attributes)
 {
     assertIsMainRunLoop();
-    workQueue().dispatch([attributes = WTFMove(attributes), protectedThis = Ref { *this }]() mutable {
-        protectedThis->workQueueInitialize(WTFMove(attributes));
+    workQueue().dispatch([attributes = WTFMove(attributes), sharedPreferencesForWebProcess = m_gpuConnectionToWebProcess.get()->sharedPreferencesForWebProcess(), protectedThis = Ref { *this }]() mutable {
```

Comment:
This function is called during object construction.
```
   : ...
    m_sharedPreferencesForWebProcess(gpuConnectionToWebProcess->sharedPreferencesForWebProcess())
```

The settings could be just read at the constructor, no need to read them after constructor, pass them in work queue callback and assign only at the work queue initialization.

Reply:
Oh, that's much cleaner! Will fix.